support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/SingleLogoutPreparationAction.java

```diff
@@ -0,0 +1,79 @@
+package org.apereo.cas.support.pac4j.web.flow;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apereo.cas.util.Pac4jUtils;
+import org.apereo.cas.web.support.CookieRetrievingCookieGenerator;
+import org.apereo.cas.web.support.WebUtils;
+import org.pac4j.core.context.J2EContext;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileManager;
+import org.pac4j.core.store.Store;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+
+/**
+ * The purpose of this action is to prepare the PAC4J Profile Manager for Single Logout.
+ * 
+ * The Profile Manager keeps the profiles in request + session but the session has already been destroyed. This action should restore the
+ * profile from a long term storage - {@link Store} and populate the PAC4J Profile Manager with it.
+ * 
+ * This action should be called from the Logout web flow.
+ * 
+ * @author jkacer
+ * 
+ * @since 5.3.0
+ */
+public class SingleLogoutPreparationAction extends AbstractAction {
+
+    private static final Logger LOGGER2 = LoggerFactory.getLogger(SingleLogoutPreparationAction.class);
+
+    private final Store<String, CommonProfile> profileStore;
+    private final CookieRetrievingCookieGenerator ticketGrantingTicketCookieGenerator;
+
+
+    public SingleLogoutPreparationAction(final CookieRetrievingCookieGenerator ticketGrantingTicketCookieGenerator,
+            final Store<String, CommonProfile> profileStore) {
+        super();
+        this.profileStore = profileStore;
+        this.ticketGrantingTicketCookieGenerator = ticketGrantingTicketCookieGenerator;
+    }
+
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
```

Comment:
Remove; fix the warnings where applicable.

Reply:
OK, will be removed on next commit.